contributors/devel/sig-scalability/kubemark-setup-guide.md

```diff
@@ -0,0 +1,71 @@
+## Introduction
+This document serves to understand how to set up kubemark cluster given that a base cluster (to run hollow-node pods) and separate master instance(s) (to act as master(s) for the hollow nodes) are already present.
+
+## Precondition
+You need two k8s clusters(kubemark cluster and external cluster) to set up a final kubemark cluster.
+
+The functions of the two clusters are as follows:
+
+- kubemark cluster: the cluster only needs masters, its masters will be the final kubemark cluster's masters
+- external cluster: it's used to create hollow nodes for the final kubemark cluster.
+
+## Steps:
+1. Build kubemark image
+
+If the latest image of `staging-k8s.gcr.io/kubemark` in the public repo is available, you can skip this section. 
+
+If the image is not available, you can build it by yourself as follows.
+
+- i. pull kubernetes code
+
+```
+cd $GOPATH/src/k8s.io/
+git clone git@github.com:kubernetes/kubernetes.git
+```
+
+- ii. build kubemark binary 
+
+```
+./hack/build-go.sh cmd/kubemark/
+cp $GOPATH/src/k8s.io/kubernetes/_output/bin/kubemark $GOPATH/src/k8s.io/kubernetes/cluster/images/kubemark/
+```
+
+- iii. build kubemark image
+
+```
+cd $GOPATH/src/k8s.io/kubernetes/cluster/images/kubemark/
+make build
+```
+
+Then you can get the image named `staging-k8s.gcr.io/kubemark:latest` locally
+
+- iv. push kubemark image
+
+```
+docker tag staging-k8s.gcr.io/kubemark:latest {{kubemark_image_registry}}/kubemark:{{kubemark_image_tag}}
+docker push {{kubemark_image_registry}}/kubemark:{{kubemark_image_tag}}
+```
+
+2. Create hollow nodes
+
+- i. create namesapce, configmap and secret
+
+Copy kubemark cluster's kubeConfig, put it on a master of external cluster, rename it as config.
+
+```
+kubectl create ns kubemark
+kubectl create configmap node-configmap -n kubemark --from-literal=content.type="test-cluster"
+kubectl create secret generic kubeconfig --type=Opaque --namespace=kubemark --from-file=kubelet.kubeconfig=config --from-file=kubeproxy.kubeconfig=config
+```
+
+- ii. use yaml to create hollow nodes
+
+You can use `hollow-node_simplified_template.yaml` in the current directory, or use `hollow-node_template.yaml` in `kubernetes/test/kubemark/resources/`.
+
+```
+kubectl create -f hollow-node_simplified_template.yaml
```

Comment:
Can you add a note below this command saying that your external cluster should have enough resources to be able to run `{{numreplicas}}` no. of hollow-node pods?

Reply:
Done